src/pages/trade/ui/route-book.tsx

```diff
@@ -1,60 +1,212 @@
+              <th className='py-[8px] font-normal text-right'>Total</th>
+              <th className='py-[8px] font-normal text-right'>Route</th>
             </tr>
-          ))}
-        </tbody>
-      </table>
+          </thead>
+
+          <tbody className='relative'>
+            {multiHops.sell.map((trace, idx) => (
+              <TradeRow
+                key={`${trace.price}-${trace.total}-${idx}`}
+                trace={trace}
+                isSell={true}
+                relativeSize={sellRelativeSizes.get(trace.total) ?? 0}
+              />
+            ))}
+
+            <SpreadRow sellOrders={multiHops.sell} buyOrders={multiHops.buy} />
+
+            {multiHops.buy.map((trace, idx) => (
+              <TradeRow
+                key={`${trace.price}-${trace.total}-${idx}`}
+                trace={trace}
+                isSell={false}
+                relativeSize={buyRelativeSizes.get(trace.total) ?? 0}
+              />
+            ))}
+          </tbody>
+        </table>
+      </div>
     </div>
   );
 });
 
 export const RouteBook = observer(() => {
-  const { data, isLoading, error } = useBook();
+  const { data: bookData, isLoading: bookIsLoading, error: bookErr } = useBook();
 
-  if (error) {
-    return <div>Error loading route book: ${String(error)}</div>;
+  if (bookIsLoading || !bookData) {
+    return <div className='text-gray-400'>Loading...</div>;
   }
 
-  if (isLoading || !data) {
-    return <RouteBookLoadingState />;
+  if (bookErr) {
+    return <div className='text-red-500'>Error loading route book: {String(bookErr)}</div>;
   }
 
-  return <RouteBookData bookData={data} />;
+  return <RouteBookData bookData={bookData} />;
 });
+
+const calculateSpread = (sellOrders: Trace[], buyOrders: Trace[]) => {
+  if (!sellOrders.length || !buyOrders.length) {
+    return null;
+  }
+
+  const lowestSell = sellOrders[sellOrders.length - 1];
+  const highestBuy = buyOrders[0];
+
+  if (lowestSell === undefined || highestBuy === undefined) {
+    return null;
+  }
+
+  const sellPrice = parseFloat(lowestSell.price);
+  const buyPrice = parseFloat(highestBuy.price);
+
+  const spread = sellPrice - buyPrice;
+  const midPrice = (sellPrice + buyPrice) / 2;
+  const spreadPercentage = (spread / midPrice) * 100;
+
+  return {
+    amount: spread.toFixed(8),
+    percentage: spreadPercentage.toFixed(2),
+    midPrice: midPrice.toFixed(8),
+  };
+};
+
+const SpreadRow = ({ sellOrders, buyOrders }: { sellOrders: Trace[]; buyOrders: Trace[] }) => {
+  const spreadInfo = calculateSpread(sellOrders, buyOrders);
+
+  if (!spreadInfo) {
+    return null;
+  }
+
+  return (
+    <tr>
+      <td colSpan={4} className='border-y border-[#262626]'>
+        <div className='flex items-center justify-center gap-2 px-3 py-3 text-xs'>
+          <span className='text-[#55D383]'>{spreadInfo.midPrice}</span>
+          <span className='text-gray-400'>Spread:</span>
+          <span className='text-white'>{spreadInfo.amount} USDC</span>
```

Comment:
issue: `USDC` should not be hardcoded right? This should probably be the quote asset symbol.